File: openapi-common-sdk-demo/net/MCatCloud/MCatCloud/Common/ClientProfile.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace MCatCloud.Common
{
    /// <summary>
    /// Client profiles.
    /// </summary>
    public class ClientProfile
    {
        /// <summary>
        /// Constructor.
        /// </summary>
        /// <param name="signMethod">Signature process method.</param>
        /// <param name="httpProfile">HttpProfile instance.</param>
        public ClientProfile(string signMethod, HttpProfile httpProfile)
        {
            this.SignMethod = signMethod;
            this.HttpProfile = httpProfile;
            this.Language = Language.DEFAULT;
        }

        public ClientProfile(string signMethod)
            : this(signMethod, new HttpProfile())
        {

        }

        public ClientProfile()
            : this(SIGN_SHA256)
        {
        }

        /// <summary>
        /// HTTP profiles, refer to <seealso cref="HttpProfile"/>
        /// </summary>
        public HttpProfile HttpProfile { get; set; }

        /// <summary>
        /// Signature process method.
        /// </summary>
        public string SignMethod { get; set; }

        /// <summary>
        /// valid choices: zh-CN, en-US
        /// </summary>
        public Language Language { get; set; }

        /// <summary>
        /// Signature process version 1, with HmacSHA1.
        /// </summary>
        public const string SIGN_SHA1 = "hmac-sha1";

        /// <summary>
        /// Signature process version 1, with HmacSHA256.
        /// </summary>
        public static string SIGN_SHA256 = "hmac-sha256";

        /// <summary>
        /// Signature process version 1, with HmacSHA512.
        /// </summary>
        public static string SIGN_SHA512 = "hmac-sha512";

   
    }
}

File: openapi-common-sdk-demo/java/MCatCloud/src/Common/ClientProfile.java
package Common;

public class ClientProfile {
    public ClientProfile(String signMethod, HttpProfile httpProfile)
    {
        this.signMethod = signMethod;
        this.httpProfile = httpProfile;
        this.language = Language.DEFAULT;
    }



    /// <summary>
    /// HTTP profiles, refer to <seealso cref="HttpProfile"/>
    /// </summary>
    public HttpProfile httpProfile ;

    /// <summary>
    /// Signature process method.
    /// </summary>
    public String signMethod;

    /// <summary>
    /// valid choices: zh-CN, en-US
    /// </summary>
    public String language;

    /// <summary>
    /// Signature process version 1, with HmacSHA1.
    /// </summary>
    public static final String SIGN_SHA1 = "hmac-sha1";

    /// <summary>
    /// Signature process version 1, with HmacSHA256.
    /// </summary>
    public static final String SIGN_SHA256 = "hmac-sha256";

    /// <summary>
    /// Signature process version 1, with HmacSHA512.
    /// </summary>
    public static final String SIGN_SHA512 = "hmac-sha512";
}
